http-client/src/main/java/io/micronaut/http/client/netty/DefaultHttpClient.java

```diff
@@ -1526,16 +1546,32 @@ private <R extends HttpResponse<?>> Flux<R> handleStreamHttpError(
     private void streamRequestThroughChannel0(
             io.micronaut.http.HttpRequest<?> parentRequest,
             final io.micronaut.http.HttpRequest<?> finalRequest,
-            FluxSink emitter,
+            FluxSink<? super MutableHttpResponse<?>> emitter,
             ConnectionManager.PoolHandle poolHandle,
             boolean secure) throws HttpPostRequestEncoder.ErrorDataEncoderException {
-        NettyRequestWriter requestWriter = prepareRequest(
-                finalRequest,
-                finalRequest.getUri(),
-                emitter
+        if (!(finalRequest instanceof MutableHttpRequest)) {
+            throw new IllegalArgumentException("A MutableHttpRequest is required");
```

Comment:
can we re-construct a mutable http request if the request is not one?

Reply:
i went through the code and changed it to use MutableHttpRequest throughout. at the edges (where the user passes in the possibly-immutable request) i added a new wrapper.